### PR TITLE
Update Github Actions

### PR DESCRIPTION
**Pull request by MSevey**
# PULL REQUEST

## Overview
Updates the github actions
1. use common hadolint
2. run lint
3. run tests
4. handle release
5. build and push docker image

## Checklist
Review and complete the checklist to ensure that the PR is complete before assigned to an approver.
 - [x] All new methods or updated methods have clear docstrings
 - [x] Testing added or updated for new methods
 - [x] Verify if any changes impact the WebPortal Health Checks
 - [x] Approriate documentation updated
 - [x] Changelog file created

## Issues Closed

Closes SKY-425

### CODE REVIEW

**Review by peterjan — APPROVED** [review]
awesome!

File: .github/workflows/ci_release.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          go-version: "1.18"
+      - name: Install analyze
+        run: go install gitlab.com/NebulousLabs/analyze@latest
+      - name: Install golangci-lint
+        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.45.0
+      - name: Lint
+        run: make lint
+      - name: Run short tests
+        run: make test
+      # TODO: enable, currently tests are broken and time out
```
Comment (peterjan) [inline .github/workflows/ci_release.yml:46]: all of the abuse related repos (and others actually) need mongo, so there's a make command called `start-mongo` that you'll probably need to run before running the tests. Is it possible that was the issue the tests appeared broken and time out

File: Makefile [hunk 2]
```diff
@@ -124,7 +114,7 @@ bench: fmt
 test:
 	go test -short -tags='debug testing netgo' -timeout=5s $(pkgs) -run=$(run) -count=$(count)
 
-test-long: lint lint-ci
+test-long: lint
 	@mkdir -p cover
 	GORACE='$(racevars)' go test -race --coverprofile='./cover/cover.out' -v -failfast -tags='testing debug netgo' -timeout=30s $(pkgs) -run=$(run) -count=$(count)
```
Comment (ro-tex) [inline Makefile:119]: This times out for me at 30s, so we should increase the timeout.

They are also failing when I do increase the timeout. @peterjan are they supposed to pass?

File: .github/workflows/ci_release.yml (hunk 1 above)
Comment (ro-tex) [reply .github/workflows/ci_release.yml:46]: We start and stop mongo when running the `test-int` tests. Maybe we should have that one here?
That being said, when I tried to do that in a GitHub action I couldn't connect to the mongo container.

**Review by ro-tex — APPROVED** [review]
Integration tests are not being run but I think that's a separate effort, so I'll approve this.

**Pull request merged**